### PR TITLE
Annot nav main update

### PR DESCRIPTION
**Pull request by Edih112**
Latest from the annotation feature. Includes the following updates:

- More detailed and correct hover information
- Grammar update allowing scientific notation (so long as the exponent sign is explicitly stated, as in '+' or '-')
- An initial working annotation feature that allows the user to search for and insert chosen annotations into the file.

### CODE REVIEW

File: src/components/antimony-editor/AntimonyEditor.tsx [hunk 1]
```diff
@@ -150,8 +263,8 @@ const AntimonyEditor: React.FC<AntimonyEditorProps & { database: IDBPDatabase<My
       const delayedModelParser = (editor: monaco.editor.IStandaloneCodeEditor) => {
         clearTimeout(typingTimer);
         typingTimer = setTimeout(() => {
-          ModelParser(editor, true);
-          ModelSemanticsChecker(editor, true);
+          // ModelParser(editor, true);
```
Comment (konankisa) [inline src/components/antimony-editor/AntimonyEditor.tsx:266]: Remove this line if it's not being used

File: src/components/antimony-editor/AntimonyEditor.tsx [hunk 2]
```diff
@@ -78,68 +80,179 @@ declare global {
...
+
+              let varInfo: Variable | undefined;
+              console.log("waa");
+
+              // check that user cursor is over an actual variable.
+              let ST = ModelSemanticsChecker(ed, false, false);
+              console.log(ST);
+              let varAndAnnotationPositionInfo = ST.hasVarAtLocation(word.word, srcRange);
+              if (varAndAnnotationPositionInfo) {
+                // alert(word.word + ", " + varInfo.idSrcRange.toString());
```
Comment (konankisa) [inline src/components/antimony-editor/AntimonyEditor.tsx:233]: Remove commented alerts if not in use

File: src/components/antimony-editor/AntimonyEditor.tsx [hunk 3]
```diff
@@ -78,68 +80,179 @@ declare global {
...
-      });
+      database
+        .transaction("files")
+        .objectStore("files")
+        .get(fileName)
+        .then((data) => {
+          if (data) {
+            // setOriginalContent(data.content);
+            setNewContent(data.content);
+            // setSelectedFile(data.name);
```
Comment (konankisa) [inline src/components/antimony-editor/AntimonyEditor.tsx:131]: Remove commented code if not in use

File: src/index.tsx [hunk 4]
```diff
@@ -30,6 +30,7 @@ const root = ReactDOM.createRoot(
 root.render(
   <React.StrictMode>
     <App/>
+    {/* <App emitter={emitter} modeldata={''}/> */}
```
Comment (konankisa) [inline src/index.tsx:33]: Remove comment if not used

File: src/language-handler/ModelSemanticChecker.ts [hunk 5]
```diff
@@ -35,19 +39,399 @@ class ErrorListener implements ANTLRErrorListener<any> {
...
+    this.parser = new AntimonyGrammarParser(tokenStream);
+
+    this.errorListener = new ErrorListener();
+    this.parser.removeErrorListeners();
+    this.parser.addErrorListener(this.errorListener);
+
+    // Parse the input, where `compilationUnit` is whatever entry point you defined
+    this.tree = this.parser.root();
+    // console.log(this.tree);
+    this.globalST = new GlobalST();
```
Comment (konankisa) [inline src/language-handler/ModelSemanticChecker.ts:134]: Remove unused console log statements

**Review by konankisa — COMMENTED** [review]
Looks good for the most part, just needs some cleanup

File: src/components/antimony-editor/AntimonyEditor.tsx (hunk 1 above)
Comment (Edih112) [reply src/components/antimony-editor/AntimonyEditor.tsx]: Done

File: src/components/antimony-editor/AntimonyEditor.tsx (hunk 3 above)
Comment (Edih112) [reply src/components/antimony-editor/AntimonyEditor.tsx]: Done

File: src/index.tsx (hunk 4 above)
Comment (Edih112) [reply src/index.tsx]: Done

File: src/language-handler/ModelSemanticChecker.ts (hunk 5 above)
Comment (Edih112) [reply src/language-handler/ModelSemanticChecker.ts:131]: Done

**Review by evaxliu — APPROVED** [review]
Tiny changes. Overall very good work.

File: src/language-handler/ModelSemanticChecker.ts [hunk 6]
```diff
@@ -63,11 +442,9 @@ export function getErrors(antimonyCode: string, includeParseErrors: boolean): Er
   // Parse the input, where `compilationUnit` is whatever entry point you defined
   let tree = parser.root();
   // printing the tree for debugging purposes
```
[the remark this answers is not in the document]
Comment (Edih112) [reply src/language-handler/ModelSemanticChecker.ts]: comment removed

File: src/components/create-annotation/CreateAnnotationModal.tsx [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+    }            
+    let text = spaces + varToAnnotate + " identity \"" + annotation.link + "\"; //" + annotation.name +"\n" ;
+
+    let selection = new monaco.Range(line, 0, line, 0);
+    let op = {identifier: id, range: selection, text: text, forceMoveMarkers: true};
+    editorInstance?.executeEdits("my-source", [op]);
+    let a: monaco.IMarkdownString;
+
+    editorInstance?.revealLineInCenter(line);
+    // TODO: try to select the new annotation to make it clear which line was added.
```
[the remark this answers is not in the document]
Comment (Edih112) [reply src/components/create-annotation/CreateAnnotationModal.tsx]: removed

File: src/components/antimony-editor/AntimonyEditor.tsx [hunk 8]
```diff
@@ -78,68 +80,172 @@ declare global {
...
         // Create the editor
         editor = monaco.editor.create(editorRef.current, {
           bracketPairColorization: { enabled: true }, // Enable bracket pair colorization
           value: content,
-          language: 'xml',
+          language: "xml",
+          automaticLayout: true,
         });
-        console.log('SBML Editor');
+        console.log("SBML Editor");
```
[the remark this answers is not in the document]
Comment (Edih112) [reply src/components/antimony-editor/AntimonyEditor.tsx]: removed

**Pull request merged**